airflow/include/tasks/extract/utils/html_utils.py

```diff
@@ -23,28 +24,74 @@ def is_valid_url(url):
     return bool(parsed.netloc) and bool(parsed.scheme)
 
 
-def exclude_path(url, exclude_docs=None):
+def fetch_page_content(url: str) -> str:
+    try:
+        response = requests.get(url)
+        response.raise_for_status()  # Raise an HTTPError for bad responses
+        return response.content
+    except requests.RequestException:
+        logger.info("Error fetching content for %s: %s", url, url)
+        return ""
+
+
+def is_excluded_url(url: str, exclude_literal: list[str]) -> bool:
+    url_path = urlparse(url).path
+    return any(literal in url_path for literal in exclude_literal)
+
+
+def clean_tags(text_content: str, tags: list[str] | None) -> str | None:
     """
-    Check if the URL path contains any of the specified strings to be excluded.
+    Clean the HTML content by removing script and style tags, collapsing whitespaces, and extracting text.
 
-    param url (str): The URL to check.
-    param exclude_docs (list): List of strings to exclude from the URL path.
+    param text_content (str): The HTML content to be cleaned.
     """
-    if exclude_docs is None:
-        exclude_docs = []
-    url_path = urlparse(url).path
-    return any(docs in url_path for docs in exclude_docs)
+    if tags is None:
+        tags = ["script", "style"]
+    soup = BeautifulSoup(text_content, "html.parser").find("body")
 
+    if soup is None:
+        return
+    # Remove script and style tags
+    for script_or_style in soup(tags):
+        script_or_style.extract()
+
+    # Get text and handle whitespaces
+    text = " ".join(soup.stripped_strings)
+
+    return text
 
-def get_all_links(url, exclude_docs=None):
+
+def truncate_tokens(text: str, encoding_name: str = "gpt-3.5-turbo", max_length: int = 8192) -> str | None:
+    """
+    Truncates a text string based on the maximum number of tokens.
+
+    param string (str): The input text string to be truncated.
+    param encoding_name (str): The name of the encoding model.
+    param max_length (int): The maximum number of tokens allowed. Default is 8192.
```

Comment:
Why is default 8192? Any idea?

Reply:
It depends on embedding model, In our case I have notice if token size is bigger than 8192 it fail